__tests__/AppCatalog.js

```diff
@@ -0,0 +1,141 @@
+import '@testing-library/jest-dom/extend-expect';
+import {
+  appCatalogsResponse,
+  appsResponse,
+  AWSInfoResponse,
+  catalogIndexResponse,
+  getPersistedMockCall,
+  getMockCall,
+  ORGANIZATION,
+  orgResponse,
+  orgsResponse,
+  releasesResponse,
+  userResponse,
+  V4_CLUSTER,
+  v4AWSClusterResponse,
+  v4AWSClusterStatusResponse,
+  v4ClustersResponse,
+} from 'testUtils/mockHttpCalls';
+import { fireEvent, wait } from '@testing-library/react';
+import { renderRouteWithStore } from 'testUtils/renderUtils';
+import nock from 'nock';
+
+// Cluster and route we are testing with.
+const ROUTE = `/app-catalogs/`;
+
+// Tests setup
+const requests = {};
+
+// Responses to requests
+beforeAll(() => {
+  // prettier-ignore
+  requests.userInfo = getPersistedMockCall('/v4/user/', userResponse);
+  requests.info = getPersistedMockCall('/v4/info/', AWSInfoResponse);
+  requests.organizations = getPersistedMockCall(
+    '/v4/organizations/',
+    orgsResponse
+  );
+  requests.organization = getPersistedMockCall(
+    `/v4/organizations/${ORGANIZATION}/`,
+    orgResponse
+  );
+  requests.clusters = getPersistedMockCall('/v4/clusters/', v4ClustersResponse);
+  requests.cluster = getPersistedMockCall(
+    `/v4/clusters/${V4_CLUSTER.id}/`,
+    v4AWSClusterResponse
+  );
+  requests.status = getPersistedMockCall(
+    `/v4/clusters/${V4_CLUSTER.id}/status/`,
+    v4AWSClusterStatusResponse
+  );
+  requests.apps = getPersistedMockCall(
+    `/v4/clusters/${V4_CLUSTER.id}/apps/`,
+    appsResponse
+  );
+  requests.credentials = getPersistedMockCall(
+    `/v4/organizations/${ORGANIZATION}/credentials/`
+  );
+  requests.releases = getPersistedMockCall('/v4/releases/', releasesResponse);
+
+  requests.appcatalogs = getPersistedMockCall(
+    '/v4/appcatalogs/',
+    appCatalogsResponse
+  );
+
+  nock('https://cors-anywhere.herokuapp.com').get(/.*$/).reply(500, 'Not supposed to get here in these tests.').persist();
```

Comment:
You can use `getPersistedMockCall` for this

Reply:
I kind of like the story this tells, that these 3 requests happen once in this test.

Reply:
I think for now I don't want to sweep under the rug that all catalog indexes are loaded, even if we just go straight to a specific catalog.

Reply:
What I meant is using `getPersistedMockCall` basically does the same thing, but in a shorter form.
Although I now realize that this returns status code `500`, which is not supported by `getPersistedMockCall`. So it will only work for the next 3 ones in `renders all non internal app catalogs in the app catalogs overview`

Reply:
Nevermind, I didn't realize `getPersistendMockCall` does not support custom domains and custom status codes.